client/src/templates/Challenges/odin/Show.tsx

```diff
@@ -0,0 +1,431 @@
+
+  handleAssignmentChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+    assignments: number
+  ): void => {
+    const assCopy = this.state.assignmentsCompleted;
+    const completed = event.target.checked ? assCopy + 1 : assCopy - 1;
+    const allCompleted = assignments == completed;
+
+    this.setState({
+      assignmentsCompleted: completed,
+      allAssignmentsCompleted: allCompleted
+    });
+  };
+
+  onVideoLoad = () => {
+    this.setState({
+      videoIsLoaded: true
+    });
+  };
+
+  render() {
+    const {
+      data: {
+        challengeNode: {
+          challenge: {
+            fields: { blockName },
+            title,
+            description,
+            superBlock,
+            certification,
+            block,
+            translationPending,
+            videoId,
+            videoLocaleIds,
+            bilibiliIds,
+            question: { text, answers, solution, assignments }
+          }
+        }
+      },
+      openCompletionModal,
+      pageContext: {
+        challengeMeta: { nextChallengePath, prevChallengePath }
+      },
+      t,
+      isChallengeCompleted
+    } = this.props;
+
+    const blockNameTitle = `${t(
+      `intro:${superBlock}.blocks.${block}.title`
+    )} - ${title}`;
+    return (
+      <Hotkeys
+        executeChallenge={() => {
+          this.handleSubmit(solution, openCompletionModal, assignments);
+        }}
+        innerRef={(c: HTMLElement | null) => (this._container = c)}
+        nextChallengePath={nextChallengePath}
+        prevChallengePath={prevChallengePath}
+      >
+        <LearnLayout>
+          <Helmet
+            title={`${blockNameTitle} | ${t('learn.learn')} | freeCodeCamp.org`}
+          />
+          <Grid>
+            <Row>
+              <Spacer />
+              <ChallengeTitle
+                isCompleted={isChallengeCompleted}
+                translationPending={translationPending}
+              >
+                {title}
+              </ChallengeTitle>
+
+              <Col lg={10} lgOffset={1} md={10} mdOffset={1}>
+                <div className='video-wrapper'>
+                  {!this.state.videoIsLoaded ? (
+                    <div className='video-placeholder-loader'>
+                      <Loader />
+                    </div>
+                  ) : null}
+                  <VideoPlayer
+                    bilibiliIds={bilibiliIds}
+                    onVideoLoad={this.onVideoLoad}
+                    title={title}
+                    videoId={videoId}
+                    videoIsLoaded={this.state.videoIsLoaded}
+                    videoLocaleIds={videoLocaleIds}
+                  />
+                </div>
+              </Col>
+              <Col md={8} mdOffset={2} sm={10} smOffset={1} xs={12}>
+                <h2>{title}</h2>
+                <PrismFormatted className={'line-numbers'} text={description} />
+                <Spacer />
+                <ObserveKeys>
+                  {assignments[0] != '' ? (
```

Comment:
Using ternaries when the or branch is `null` isn't a standard React pattern.  In this cases it's customary to use `{ condition && (`, since you achieve the same effect with cleaner code.